src/navigation/AppNavigator.tsx

```diff
@@ -0,0 +1,36 @@
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import React, { useEffect, useState } from 'react';
+import { APP_NAME, SPLASH_SCREEN_TIME } from '../constants/AppConstant';
+import Homepage from '../screens/homepage';
+import SplashScreen from '../screens/splash';
+
+const Stack = createNativeStackNavigator();
+
+const AppNavigator = () => {
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        setTimeout(() => {
+            setLoading(false);
+        }, SPLASH_SCREEN_TIME);
+    }, []);
+
+    return (
+        <Stack.Navigator
+            initialRouteName={loading ? APP_NAME.SPLASH : APP_NAME.HOME}
+            screenOptions={{
+                animationTypeForReplace: 'push',
+            }}>
+            {
+                loading ?
+                    <Stack.Screen options={{
```

Comment:
Ok in expo we do get the initial splash screen that we can change. But in react-native cli I didn't get anything like that. Will look into it. Thanks.

Reply:
I checked how to add a splash screen manually. Got to know something about package react-native-splash-screen which is leading to a lot of java and android code getting into play because of that. Not have much knowledge of android stuff which is leading to the break of the application build. I think splash screen is good as we can take care of ios and android in same screen.😅